src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/PrefetchGetRecordsCache.java

```diff
@@ -110,9 +121,11 @@ public void shutdown() {
     }
 
     private class DefaultGetRecordsCacheDaemon implements Runnable {
+        private volatile boolean isShutdown = false;
+        
         @Override
         public void run() {
-            while (true) {
+            while (!isShutdown) {
                 if (Thread.currentThread().isInterrupted()) {
                     log.warn("Prefetch thread was interrupted.");
                     callShutdownOnStrategy();
```

Comment:
This can be moved outside the while loop.

Reply:
Moving it to outside the loop will just check it at the start, I had kept it inside the loop just for the case if it the thread gets the interrupt at that point.